SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorModule.cpp

```diff
@@ -82,6 +85,63 @@ bool FSpatialGDKEditorModule::CanExecuteLaunch() const
 	return SpatialGDKEditorInstance->GetPackageAssemblyRef()->CanBuild();
 }
 
+bool FSpatialGDKEditorModule::CanStartSession() const
+{
+	if (!SpatialGDKEditorInstance->IsSchemaGenerated())
+	{
+		EAppReturnType::Type Result = FMessageDialog::Open(EAppMsgType::Ok, LOCTEXT("MissingSchema", "Attempted to start a local deployment but schema is not generated. You can generate it by clicking on the Schema button in the toolbar."));
+
+
+		return false;
+	}
+
+	if (ShouldConnectToCloudDeployment())
+	{
+		if (GetDevAuthToken().IsEmpty())
+		{
+			FMessageDialog::Open(EAppMsgType::Ok, LOCTEXT("MissingDevelopmentAuthenticationToken", "You have to generate or provide a development authentication token in the SpatialOS GDK Editor Settings section to enable connecting to a cloud deployment."));
+			return false;
+		}
+
+		const USpatialGDKEditorSettings* Settings = GetDefault<USpatialGDKEditorSettings>();
+		bool bIsRunningInChina = GetDefault<USpatialGDKSettings>()->IsRunningInChina();
+		FText OutErrorMessage;
+		if (!SpatialCommandUtils::HasDevLoginTag(Settings->DevelopmentDeploymentToConnect, bIsRunningInChina, OutErrorMessage))
```

Comment:
I would slightly prefer to move the empty string check for `DevelopmentDeploymentToConnect` here and only call `HasDevLoginTag` if a deployment is specified.

Reply:
hm.. I actually prefer it to be in `HasDevLoginTag`. While it's currently only being called here, it is a public function and could be called from anywhere. I think it's good to check at the start whether a name has actually been passed in.

Reply:
Right, but at the moment, it returns `true` if an empty deployment name is passed in, which is a special case here and would be wrong in general.

Reply:
fair enough. in which case I'll change the check to return false and add a special case to not call this function if no name has specified.